src/olympia/promoted/cron.py

```diff
@@ -0,0 +1,41 @@
+import olympia.core.logger
+from olympia import amo
+from olympia.addons.models import Addon
+from olympia.constants.promoted import NOTABLE
+from olympia.promoted.models import PromotedAddon
+from olympia.zadmin.models import get_config
+
+
+log = olympia.core.logger.getLogger('z.promoted.cron')
+
+ADU_LIMIT_CONFIG_KEY = 'notable-adu-threshold'
+
+
+def add_high_adu_extensions_to_notable():
+    """Add add-ons with high ADU to Notable promoted group."""
+
+    adu_limit = get_config(ADU_LIMIT_CONFIG_KEY)
+    if not adu_limit:
+        log.info('[%s] config key not set', ADU_LIMIT_CONFIG_KEY)
+        return
+    addons_ids_and_slugs = (
+        Addon.objects.public()
+        .filter(
+            average_daily_users__gte=adu_limit,
+            type=amo.ADDON_EXTENSION,
+            promotedaddon=None,
+        )
+        .values_list('id', 'slug', 'average_daily_users')
+    )
+    count = len(addons_ids_and_slugs)
+    log.info('Starting adding %s addons to %s', count, NOTABLE.name)
+    for addon_id, addon_slug, adu in addons_ids_and_slugs:
+        log.info(
+            'Adding addon id[%s], slug[%s], with ADU[%s] to %s.',
+            addon_id,
+            addon_slug,
+            adu,
+            NOTABLE.name,
+        )
+        PromotedAddon.objects.create(addon_id=addon_id, group_id=NOTABLE.id)
```

Comment:
To confirm: a race condition of an admin manually creating a `PromotedAddon` around the same time the cron/task runs?  Or some other scenario?

Reply:
Any promotion created at the same time for that add-on, yeah. Unlikely, but annoying because with the way it's written right now it'd kill the entire script.

Reply:
True, but worst case with an error it'd try again the next day.

Reply:
I'm likely going to change it to `get_or_create` anyway to address Andreas's comment about wanting Not Promoted add-ons to be updated. That highlights the opposite problem - that we don't really _want_ a `Notable` group add-on to override a manual curation.

Reply:
https://github.com/mozilla/addons-server/pull/19999/commits/081a29acba3054fd9ba15811d4e9ea073e986153 does this (among other stuff) - inside the loop there's an extra checks that the group is actually `Not Promoted` before updating it too.